sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphPemCertificateList.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Azure.Media.Analytics.Edge.Models
+{
+    /// <summary>
+    /// Initializes a new instance of the MediaGraphPemCertificateList class.
+    /// </summary>
+    public partial class MediaGraphPemCertificateList
+    {
+        /// <summary>
+        /// How long the base 64 lines should be.
+        /// </summary>
+        private const int Base64LineLength = 76;
+
+        /// <summary>
+        /// Header for a PEM formatted certificate.
+        /// </summary>
+        /// <remarks>
+        /// See https://tools.ietf.org/html/rfc7468 Section 5.1.
+        /// </remarks>
+        private const string PemCertificateHeader = "-----BEGIN CERTIFICATE-----";
+
+        /// <summary>
+        /// Footer for a PEM formatted certificate.
+        /// </summary>
+        /// <remarks>
+        /// See https://tools.ietf.org/html/rfc7468 Section 5.1.
+        /// </remarks>
+        private const string PemCertificateFooter = "-----END CERTIFICATE-----";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaGraphPemCertificateList"/> class.
+        /// </summary>
+        /// <param name="certificates"> The list of certificates.</param>
+        public MediaGraphPemCertificateList(IList<X509Certificate2> certificates)
+        {
+            if (certificates == null)
+            {
+                throw new ArgumentNullException(nameof(certificates));
+            }
+
+            Certificates = certificates.Select(ConvertToPemString).ToList();
+            Type = "#Microsoft.Media.MediaGraphPemCertificateList";
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaGraphPemCertificateList"/> class.
+        /// </summary>
+        /// <param name="certificates"> The certificates params.</param>
+        public MediaGraphPemCertificateList(params X509Certificate2[] certificates)
+        {
+            if (certificates == null)
+            {
+                throw new ArgumentNullException(nameof(certificates));
+            }
+
+            Certificates = certificates.Select(ConvertToPemString).ToList();
+            Type = "#Microsoft.Media.MediaGraphPemCertificateList";
+        }
+
+        private static string ConvertToPemString(X509Certificate2 certificate)
```

Comment:
Do we have a test for this method?

Reply:
The tests for this are currently in the service repo. We have plans to move some of them to this repo in the future.